src/ArchitectNow.Web/WebStarter.cs

```diff
@@ -50,28 +51,34 @@ public int Run(string[] args, Action<LoggerConfiguration> configureLogger)
             }
         }
 
-        private IWebHost BuildWebHost(string[] args)
+        private IHost BuildWebHost(string[] args)
         {
             var webHostBuilder = CreateWebHostBuilder(args);
             return webHostBuilder
                 .Build();
         }
 
-        protected virtual IWebHostBuilder CreateWebHostBuilder(string[] args)
+        protected virtual IHostBuilder CreateWebHostBuilder(string[] args)
         {
             //pulls in environment from cli args or env vars
-            var builder = new ConfigurationBuilder()
+            var configBuilder = new ConfigurationBuilder()
                 .AddEnvironmentVariables()
                 .AddCommandLine(args);
 
-            var configuration = builder.Build();
+            var configuration = configBuilder.Build();
 
             //https://github.com/aspnet/MetaPackages/blob/633cb681493c0958a9d215624c173db29e20c23d/src/Microsoft.AspNetCore/WebHost.cs
 
-            var webHostBuilder = WebHost.CreateDefaultBuilder(args)
-                .UseConfiguration(configuration)
-                .UseStartup<TStartup>()
-                .UseSerilog(Log.Logger);
+            // Using generic Host here because it allows us to inject a custom service provider for DI
+            var webHostBuilder = Host.CreateDefaultBuilder(args)
+                .UseServiceProviderFactory(new AutofacServiceProviderFactory())
```

Comment:
https://autofaccn.readthedocs.io/en/latest/integration/aspnetcore.html#asp-net-core-3-0-and-generic-hosting